hydra-haskell/src/main/haskell/Hydra/Impl/Haskell/Sources/Ext/Shacl/Model.hs

```diff
@@ -32,27 +38,43 @@ shaclModel = Graph shaclModelName elements (const True) hydraCoreName
             list $ shacl "Shape",
 
           field "closed" $
-            doc "See https://www.w3.org/TR/shacl/#ClosedConstraintComponent"
-            boolean,
+            doc "See https://www.w3.org/TR/shacl/#ClosedConstraintComponent" $
+            optional $ shacl "Closed",
+
+          field "class" $
+            doc "See https://www.w3.org/TR/shacl/#ClassConstraintComponent" $
+            set $ element $ rdf "RdfsClass",
 
           field "datatype" $
             doc "See https://www.w3.org/TR/shacl/#DatatypeConstraintComponent" $
-            set $ rdf "Iri",
+            optional $ rdf "Iri",
+
+          field "disjoint" $
+            doc "See https://www.w3.org/TR/shacl/#DisjointConstraintComponent" $
+            set $ element $ rdf "Property",
+
+          field "equals" $
+            doc "See https://www.w3.org/TR/shacl/#EqualsConstraintComponent" $
+            set $ element $ rdf "Property",
 
           field "hasValue" $
             doc ("Specifies the condition that at least one value node is equal to the given RDF term. " ++
                  "See https://www.w3.org/TR/shacl/#HasValueConstraintComponent") $
             set $ rdf "Node",
 
-          field "ignoredProperties" $
-            doc "See https://www.w3.org/TR/shacl/#ClosedConstraintComponent" $
-            list $ element $ rdf "Property",
-
           field "in" $
             doc ("Specifies the condition that each value node is a member of a provided SHACL list. " ++
                  "See https://www.w3.org/TR/shacl/#InConstraintComponent") $
             list $ rdf "Node",
 
+          field "languageIn" $
+            doc "See https://www.w3.org/TR/shacl/#LanguageInConstraintComponent" $
+            list $ string,
```

Comment:
The `rdf:LanguageTag` type can be used here. It is just an alias for a string, but in the future, it could be made into a dependent type (e.g. having a regex constraint).

Also, it appears that `languageIn` can specify a set of lists: "Each value of sh:languageIn in a shape is a [SHACL list](https://www.w3.org/TR/shacl/#dfn-shacl-list)"

Reply:
Btw. I take my second comment back. SHACL does appear to allow multiple lists, but I'm not sure they would be meaningful. A single list probably makes more sense, though we might want to make it an `optional list` so we don't have to assume that an empty list means "any language is accepted"; an empty list would more properly mean "no language is accepted".

Reply:
good catch, typically all lists should be optional but only this and sh:in change the meaning